### ISSUE TITLE
improvement for rhythm check

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#352]
I have an idea for a slight improvement of the rhythm check:
actually measures that contain too many ticks are marked by colouring these measures in rose, which is very helpful to find errors (missing flags, head dots etc.).

But there is no hint if ticks are missing. Which is also ok, in many cases, esp. in case of measures containing more than one voice, because often note heads and rests are partly common.
But on the other hand, sometimes errors keep undetected and are only seens after import to the notation program.
My idea would be a sort of - maybe optional - hinting to such errors in rhythm check, too. It could be a colouring similar to the rose for "too many ticks", let's say just a light gray instead.

**Comment by hbitteur** [comment]
With many short-living voices that come and go, the downside of this may be a high number of false positives. And a too high number of false alerts does not help the end-user.

But this is something we can easily implement, using something like: voice ending before the measure end. With the ability of switching on/off the option to let the end-user decide on the score at hand.

**Comment by Bacchushlg** [comment]
I'd like to ask to provide this feature once more. There are a couple of situations where I think that it can help a user to recognise problems:
- common rests for voices (if not recognised, notes of a voice a shifted to wrong tick positions)
- erroneously recognised triplets
- missing augm. dots
- completely missing chords (ev. in one voice only)
- missing chord in one voice

As mentioned above, the feature should be as option. Maybe it can just be a function that can be accessed by the menu, e.g. "advanced rhythm check".

Also it may make sense to provide 2 versions of the check function: 
- simple check, ignoring mismatches between voices
- full check with voices check
This last, because it is rather common that the voices of a score are not completely notated. Esp. in case of piano scores, a second voice appears only between the other notes if necessary (to be more exact: these cases don't describe a real 2nd voice - the usage of voices is an abuse because there is no other way to notate such notes in "normal" notation programs).

**Comment by hbitteur** [comment]
Ii don't know what to do

**Comment by Bacchushlg** [comment]
actually rhythm check only checks, if all voices fit into a measure. I marks a measure in rose, if it does not fit.
But: it allows to have less notes than necessary.
This makes sense in many cases, esp. for piano scores where a "second voice" may appear for part of a measure (in fact, this second voice is just crutch...).
But there are many cases where you should expect that the sum of note lengths fits exactly the measure, esp. in scores where each note line contains just one voice. In these cases, the rhythm check should check for an exact match.

The feature for this exact match should be available from the main menu, maybe also with a short-cut (F?, what is free)

**Comment by hbitteur** [comment]
If I understand correctly this option:
1. Flag any voice that starts after measure start
2. Flag any voice that stops before measure stop

Problem with second item is that some measures are legally shorter than expected (case of repetition for example)

Let's consider this option as just a "display" option, with no impact on the exported material, but only intended to call user attention on some questionable measures. Then the user can set the option off, when he decides that things are now correct.

Additional question: Should we display the questionable measure in specific color or just the questionable voice?

**Comment by Bacchushlg** [comment]
Yes, it is only meant as display option, just giving the user a hint the there might be something wrong.

About the last question: if possible, only the voice. But it would be ok the mark the complete measure, too. About the colour: maybe the same as for the normal check, or something similar

**Comment by hbitteur** [comment]
Happy new year Baruch!
I'm about to review the RHYTHMS step for several reasons, one being that for unpitched percussion the notion of a voice going up and down the score gets obsolete.
By the way, I noticed these scores make a heavy use of rest "shared" among various voices. This is an additional reason.
So, you may reasonably expect that your long-awaited "short voices" option will soon be implemented using the same "wagon".  :-)
